src/webgpu/api/validation/queue/destroyed/buffer.spec.ts

```diff
@@ -0,0 +1,125 @@
+export const description = `
+Tests using a destroyed buffer on a queue.
+
+TODO:
+- test renderPass/renderBundle (setVertexBuffer, setIndexBuffer)
+- test renderPass (resolveQuerySet)
+- test renderPass/computePass (setBindGroup)
+`;
+
+import { makeTestGroup } from '../../../../../common/framework/test_group.js';
+import { ValidationTest } from '../../validation_test.js';
+
+export const g = makeTestGroup(ValidationTest);
+
+g.test('writeBuffer')
+  .desc(
+    `
+Tests that use a destroyed buffer in writeBuffer fails.
+- x= {destroyed, not destroyed (control case)}
+  `
+  )
+  .paramsSubcasesOnly(u => u.combine('buffer', ['valid', 'destroyed'] as const))
+  .fn(t => {
+    const buffer = t.createBufferWithState(t.params.buffer, {
+      size: 4,
+      usage: GPUBufferUsage.COPY_DST,
+    });
+    const shouldError = t.params.buffer === 'destroyed';
+    t.expectValidationError(() => t.queue.writeBuffer(buffer, 0, new Uint8Array(4)), shouldError);
+  });
+
+g.test('copyBufferToBuffer')
+  .desc(
+    `
+Tests that use a destroyed buffer in copyBufferToBuffer fails.
+- x= {not destroyed (control case), src destroyed, dst destroyed}
+  `
+  )
+  .paramsSubcasesOnly(u => u.combine('invalidBufferIndex', [-1, 0, 1] as const))
```

Comment:
So for this case (and the parallel texture one), I actually do think using something like:
`.combine('destroyed', ['none', 'src', 'dst', 'both'])`
would make the test names more clear as to what is happening. (Also adding the `both` case is probably a good idea.)

Reply:
done